packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -276,45 +270,42 @@ function mapValueWithFallbacks(
 				return value;
 			}
 		}
-		default:
+		case "string":
+		// TODO:
+		// This should detect invalid strings. Something like @stdlib/regexp-utf16-unpaired-surrogate could be used to do this.
+		// See SchemaFactory.string for details.
+		case "boolean":
 			return value;
+		case "object": {
+			if (value === null || isFluidHandle(value)) {
+				return value;
+			}
+		}
+		default:
+			throw new TypeError(`Received unsupported leaf value: ${value}.`);
 	}
 }
 
 /**
  * Transforms data under an Array schema.
  * @param data - The tree data to be transformed.
  * @param allowedTypes - The set of types allowed by the parent context. Used to validate the input tree.
- * @param schemaValidationPolicy - The stored schema and policy to be used for validation, if the policy says schema
- * validation should happen. If it does, the input tree will be validated against this schema + policy, and an error will
- * be thrown if the tree does not conform to the schema. If undefined, no validation against the stored schema is done.
  */
-function arrayToMapTreeFields(
-	data: Iterable<InsertableContent>,
+function arrayChildToMapTree(
```

Comment:
I think its much cleaner to move the looping over the data into the call-site (and have it use Array.from, which could also be more efficient). If this had tests it would make targeted testing of this policy easier.